webserver.py

```diff
@@ -41,61 +42,178 @@ def get_conn():
+
+    @staticmethod
+    def new(slug, password):
+        game_state = GameState.load(slug, password)
+        if game_state:
+            raise Forbidden("Username already taken")
+        else:
+            game_state = GameState({
+                'resources': {},
+                'slug': slug,
+                'password': password,
+                'unlockCount': 0,
+                'seedCounts': {},
+                'recipes': ['a']
+            })
+        return game_state.initialize()
+
+    @classmethod
+    def load(cls, slug, password):
+        my_conn = get_conn()
+        with my_conn:
+            with my_conn.cursor() as cursor:
+                cursor.execute('SELECT game_state FROM game_states WHERE slug = %s;', [slug])
+                row = cursor.fetchone()
+                if not row:
+                    return None
+                game_state = GameState(row[0])
+                game_state.check_password(password)
+                return game_state.initialize()
+
+    def save(self, slug):
+        my_conn = get_conn()
+        json_game_state = json.dumps(self.data, indent=4)
+        with my_conn:
+            with my_conn.cursor() as cursor:
+                cursor.execute('INSERT INTO game_states (slug, game_state) VALUES (%s, %s) '
+                               'ON CONFLICT (slug) DO UPDATE SET game_state=%s;',
+                               [slug, json_game_state, json_game_state])
+
+    # add missing seed counts, resources, and plots to game_state
+    # unlock first four plots
+    # check to never run out of money
+    def initialize(self):
+        for seed_id in GAME_CONFIG['seeds']:
+            if self.data['seedCounts'].get(seed_id) is None:
+                self.data['seedCounts'][seed_id] = 0
+        for resource in GAME_CONFIG['starting_resources']:
+            if self.data['resources'].get(resource) is None:
+                self.data['resources'][resource] = GAME_CONFIG['starting_resources'][resource]
+        for i in range(GAME_CONFIG['field_width']):
+            for j in range(GAME_CONFIG['field_height']):
+                if self.data.get('plot' + str(i) + '_' + str(j)) is None:
+                    self.data['plot' + str(i) + '_' + str(j)] = {'seedType': 0, 'sowTime': 0, 'locked': 1}
+        for i in range(2):
+            for j in range(2):
+                self.get_plot(i, j)['locked'] = 0
+        return self
+
+    def check_cash(self):
+        total_seed_count = 0
+        total_crops_planted = 0
+        for seed_id in GAME_CONFIG['seeds']:
+            total_seed_count += self.data['seedCounts'][seed_id]
+        for i in range(GAME_CONFIG['field_width']):
+            for j in range(GAME_CONFIG['field_height']):
+                if self.has_planted_crop(i, j):
+                    total_crops_planted += 1
+        if total_seed_count == 0 and total_crops_planted == 0:
+            if self.data['resources']['cash'] < RECIPE_CONFIG['recipes']['a']['cashCost']:
```

Comment:
rather than using `'a'` here, check cash against the cost whatever's set as `firstSeed` in the config.  This way we're not tied to always having a potato if we want to change the planting list